llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -3360,8 +3360,16 @@ bool AddressSanitizer::instrumentFunction(Function &F,
   if (F.getLinkage() == GlobalValue::AvailableExternallyLinkage) return false;
   if (!ClDebugFunc.empty() && ClDebugFunc == F.getName()) return false;
   if (F.getName().starts_with("__asan_")) return false;
-  if (F.getName().contains("__sycl_service_kernel__"))
-    return false;
+
+  if (TargetTriple.isSPIR()) {
+    if (F.getName().contains("__sycl_service_kernel__"))
```

Comment:
@AllanZyne the name check might not be accurate, what if there is a user function that is xx__sycl_service_kernel__xx?

Reply:
Okay. 
As far as I know, currently this kernel naming style is only used for assert.

llvm/sycl/unittests/assert/assert.cpp:55